zerver/views/auth.py

```diff
@@ -1029,6 +1030,28 @@ def json_fetch_api_key(
 logout_then_login = require_post(django_logout_then_login)
 
 
+@require_post
+def logout_view(request: HttpRequest, **kwargs: Any) -> HttpResponse:
+    realm = RequestNotes.get_notes(request).realm
+    assert realm is not None
+
+    if not settings.SAML_ENABLE_SP_INITIATED_SINGLE_LOGOUT or not saml_auth_enabled(realm):
```

Comment:
How does this setting interact with multiple SAML configurations? Should this option be in the SAML configuration dictionaries rather than a global setting?

Reply:
Hmm, good thought, this was actually easy to make more flexible to suit zulipchat.com. I pushed a new with a change to that, to avoid complicating the original commit more. Here's the commit message which elaborates on the reasoning here:
```
    This gives more flexibility on a server with multiple organizations and
    SAML IdPs. Such a server can have some organizations handled by IdPs
    with SLO set up, and some without it set up. In such a scenario, having
    a generic True/False server-wide setting is insufficient and instead
    being able to specify the orgs for SLO is needed.
```